libraries/mbed/targets/hal/TARGET_Freescale/TARGET_KLXX/TARGET_KL25Z/serial_api.c

```diff
@@ -78,7 +79,17 @@ void serial_init(serial_t *obj, PinName tx, PinName rx) {
     }
     // Disable UART before changing registers
     obj->uart->C2 &= ~(UARTLP_C2_RE_MASK | UARTLP_C2_TE_MASK);
-    
+
+    // Enable UART transmitter to ensure TX activity is finished
+    obj->uart->C2 |= UARTLP_C2_TE_MASK;
+
+    // Wait for TX activity to finish
+    while(!(obj->uart->S1 & UARTLP_S1_TC_MASK));
+
+    // Disbale UARTs again
+    obj->uart->C2 &= ~(UARTLP_C2_RE_MASK | UARTLP_C2_TE_MASK);
+
```

Comment:
I suppose it could. Although it's also not a bad idea to have it in init, in case the transmitter gets in a funky state somehow. That way you know that after you init, you're good to go.